flatten.sh

```diff
@@ -1,33 +1,33 @@
 #!/usr/bin/env bash
 set -e
 
-mkdir -p flattened-contracts/functions
-
-FILES1=./contracts/*.sol
-FILES2=./contracts/library/*.sol
-FILES3=./contracts/oracles/*.sol
+FILES1=./contracts/core/*/*/*.sol
+FILES2=./contracts/core/*/*/interfaces/*.sol
+FILES3=./contracts/libraries/*/*/*.sol
 
 for f in $FILES1
 do
     filepath=${f:2}
     filename=${filepath:10}
-    echo "Flattening: $filename"
-    hevm flatten --source-file $filepath | sed '/SPDX-License-Identifier/d' | sed -e '1s/^/\/\/ SPDX-License-Identifier:  AGPL-3.0-or-later /' | sed -r 'N;s/(\/\/\/\/\/\/ .*)\n\s*$/\1/;P;D' > "flattened-contracts/$filename"
-    cat "flattened-contracts/$filename" | grep -E ") external |) public" > "flattened-contracts/functions/$filename"
+    mkdir -p "flattened-contracts/${filename%/*}"
```

Comment:
Why are we making new directories for each iteration in the for loop?

Reply:
hevm flattener requires the output directory to exist, as it will only create directories within it.

If you remove that line, you get:
`./flatten.sh: line 14: flattened-contracts/core/chainlink-oracle/v1/ChainlinkOracle.sol: No such file or directory`